src/canvas-driver.js

```diff
@@ -1,148 +1,255 @@
-function renderElement (context, element, parent) {
-  if (!element) {
-    return;
-  }
-
-  if (!parent) {
-    parent = {x: 0, y: 0};
+function flatten (array) {
+  if (typeof array.reduce !== 'function') {
+    return array;
```

Comment:
This code is a bit misleading. The if statement is there because this flatten is recursive. So if you call flatten on something that's not an array, it just gets returned. Probably better to replace this conditional with !Array.isArray